app/controllers/user_badges_controller.rb

```diff
@@ -58,6 +58,11 @@ def create
     post_id = nil
 
     if params[:reason].present?
+      unless is_badge_reason_valid? params[:reason]
+        render json: { failed: 'External link not allowed in reason' }, status: 400
```

Comment:
You can combine the `return` and the `render` ;)

```suggestion
        return render json: { failed: 'External link not allowed in reason' }, status: 400
```

Reply:
The text of the error should be localized (ie. use `I18n.t("name of the string")`)